hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetAsyncDiskService.java

```diff
@@ -167,18 +167,26 @@ synchronized long countPendingDeletions() {
    * Execute the task sometime in the future, using ThreadPools.
    */
   synchronized void execute(FsVolumeImpl volume, Runnable task) {
-    if (executors == null) {
-      throw new RuntimeException("AsyncDiskService is already shutdown");
-    }
-    if (volume == null) {
-      throw new RuntimeException("A null volume does not have a executor");
-    }
-    ThreadPoolExecutor executor = executors.get(volume.getStorageID());
-    if (executor == null) {
-      throw new RuntimeException("Cannot find volume " + volume
-          + " for execution of task " + task);
-    } else {
-      executor.execute(task);
+    try {
```

Comment:
If the task is executed, the reference is closed at the end of ReplicaFileDeleteTask#run().
This part of the code handles the case when the task is not executed that we have to close it explicitly.

It might be a good idea to turn the cleanup code ReplicaFileDeleteTask#run()
`      IOUtils.cleanupWithLogger(null, volumeRef);`
to the same style as ReplicaLazyPersistTask#run()
`try (FsVolumeReference ref = volumeRef) {`
in case that ReplicaFileDeleteTask#run() throws an exception in the middle and doesn't close the reference.

Reply:
The clean up code is in the finally block, so it will be executed even if an exception occurs. Thanks for your suggestions, I will fix it to make the style consistent.